### PR TITLE
EAR-1385-change-format-for-months-on-dates-to-be-text

### PR DESCRIPTION
**Pull request by ashtuck**
What is the context of this PR?

Changes to how the `month` input is handled in GCP. In GCP, the dummy input size is smaller and the input is now handled as a standard input, not the dropdown. Also in GCP, the dropdown for Date Type is `mm` and not Month.

Updated tests to match the react testing library methodology.

How to review

Launch Author as normal using `yarn start`. 

* Date month dummy input should be larger.
* Date month input should be a drop down.
* Date month Date Type is `Month`

Launch author using `REACT_APP_FEATURE_FLAGS="hub" yarn start`.

* Date month dummy input should be smaller.
* Date month input should be a standard input.
* Date month Date Type is `mm`

What to do after everything is green
    *   Bring this branch up-to-date with Origin/Master
    *   If there are a lot of commits or some are not helpful to read, squash them down
    *   Click the Merge button on this pull request
    *   Does this change mean the Capability examples questionnaire should be updated?
    *   Move the Jira ticket for this task into the next stage of the process

### CODE REVIEW

File: eq-author/src/App/page/PropertiesPanel/GroupedAnswerProperties/AnswerProperties/Properties/DateFormat.js [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
+        </Select>
+
+      : 
+        <Select data-test="select" value={value} onChange={onChange} id={id} name={id}>
+          <option data-test="day-month-year" value="dd/mm/yyyy">dd / Month / yyyy</option>
+          <option data-test="month-year" value="mm/yyyy">Month / yyyy</option>
+          <option data-test="year" value="yyyy">yyyy</option>
+        </Select>
+      }
+
```
Comment (tbaggaley) [inline eq-author/src/App/page/PropertiesPanel/GroupedAnswerProperties/AnswerProperties/Properties/DateFormat.js:24]:
```suggestion
      <Select data-test="select" value={value} onChange={onChange} id={id} name={id}>
          <option data-test="day-month-year" value="dd/mm/yyyy">dd / {enableOn(["hub"]) ? "Month" : "mm"} / yyyy</option>
          <option data-test="month-year" value="mm/yyyy">{enableOn(["hub"]) ? "Month" : "mm"} / yyyy</option>
          <option data-test="year" value="yyyy">yyyy</option> 
      </Select>
```

Does this work? (or better - move out into e.g.

```javascript
const monthText = enableOn(["hub]) ? "Month" : "mm";
```

and use that instead

Comment (ashtuck) [reply eq-author/src/App/page/PropertiesPanel/GroupedAnswerProperties/AnswerProperties/Properties/DateFormat.js:24]: Ah great shout. I should've picked that up on my refinement pass.

Comment (ashtuck) [reply eq-author/src/App/page/PropertiesPanel/GroupedAnswerProperties/AnswerProperties/Properties/DateFormat.js]: That's been changed.

**Review by martyncolmer — DISMISSED** [review]

**Review by martyncolmer — APPROVED** [review]

**Review by martyncolmer — DISMISSED** [review]

File: eq-author/src/components/preview/Answers/DateAnswer.js [hunk 2]
```diff
@@ -21,16 +21,16 @@ const DateField = styled.div`
...
 
 const MonthDateField = styled(DateField)`
   flex: 2;
-  max-width: 15em;
+  max-width: ${props => props.enableOn ? '5em' : '15em'};
 `;
 
 const YearDateField = styled(DateField)`
-  max-width: 6em;
+  max-width: ${props => props.enableOn ? '6em' : '6em'};
```
Comment (farres1) [inline eq-author/src/components/preview/Answers/DateAnswer.js:33]:
```suggestion
  max-width: 6em;
```

**Review by farres1 — COMMENTED** [review]
I really like the change to RTL in the tests!

File: eq-author/src/components/preview/Answers/DateAnswer.js (hunk 2 above)
Comment (ashtuck) [reply eq-author/src/components/preview/Answers/DateAnswer.js:33]: Ah didn't catch that. Great spot. Made the change.

**Review by farres1 — APPROVED** [review]

**Review by martyncolmer — APPROVED** [review]

**Pull request merged**